tests/hooks/test_http_hook.py

```diff
@@ -0,0 +1,213 @@
+    from unittest import mock
+except ImportError:
+    try:
+        import mock
+    except ImportError:
+        mock = None
+
+
+def get_airflow_connection(conn_id=None):
+    real = BaseHook('')
+    real.get_connection = mock.MagicMock()
+    conn = real.get_connection()
+    conn.host = 'http://test'
+    conn.port = 8080
+    conn.login = 'test'
+    conn.password = 'test'
+    conn.extra = '{"bareer": "test"}'
+    conn.schema = None
+    return conn
+
+
+def get_airflow_connection_noheaders(conn_id=None):
+    real = BaseHook('')
+    real.get_connection = mock.MagicMock()
+    conn = real.get_connection()
+    conn.host = 'http://test'
+    conn.port = 8080
+    conn.login = 'test'
+    conn.password = 'test'
+    conn.extra = None
+    conn.schema = None
+    return conn
+
+
+class TestHttpHook(unittest.TestCase):
+    """Test get, post and raise_for_status"""
+    def setUp(self):
+        session = requests.Session()
+        adapter = requests_mock.Adapter()
+        session.mount('mock', adapter)
+        self.get_hook = HttpHook(method='GET')
+        self.post_hook = HttpHook(method='POST')
+
+    @requests_mock.mock()
+    def test_raise_for_status_with_200(self, m):
+
+        m.get(
+            'http://test:8080/v1/test',
+            status_code=200,
+            text='{"status":{"status": 200}}',
+            reason='OK'
+        )
+        with mock.patch(
+            'airflow.hooks.base_hook.BaseHook.get_connection',
+            side_effect=get_airflow_connection
+        ):
+
+            resp = self.get_hook.run('v1/test')
+            self.assertEquals(resp.text, '{"status":{"status": 200}}')
+
+    @requests_mock.mock()
+    def test_get_request_do_not_raise_for_status_if_check_response_is_false(self, m):
+
+        m.get(
+            'http://test:8080/v1/test',
+            status_code=404,
+            text='{"status":{"status": 404}}',
+            reason='Bad request'
+        )
+
+        with mock.patch(
+            'airflow.hooks.base_hook.BaseHook.get_connection',
+            side_effect=get_airflow_connection
+        ):
+            resp = self.get_hook.run('v1/test', extra_options={'check_response': False})
+            self.assertEquals(resp.text, '{"status":{"status": 404}}')
+
+    @requests_mock.mock()
+    def test_hook_contains_header_from_extra_field(self, m):
+        with mock.patch(
+            'airflow.hooks.base_hook.BaseHook.get_connection',
+            side_effect=get_airflow_connection
+        ):
+            expected_conn = get_airflow_connection()
+            conn = self.get_hook.get_conn()
+            self.assertDictContainsSubset(json.loads(expected_conn.extra), conn.headers)
+            self.assertEquals(conn.headers.get('bareer'), 'test')
+
+    @requests_mock.mock()
+    def test_hook_uses_provided_header(self, m):
+        with mock.patch(
+            'airflow.hooks.base_hook.BaseHook.get_connection',
+            side_effect=get_airflow_connection_noheaders
+        ):
```

Comment:
@ashb I did this because the default "http_default" connection points to google.com and did not want to create a dependency on the input value of the connection. But if this looks really far too complex for you I'll change it

Reply:
Since we are mocking the requests here anyway, I think it’s cleaner to just use that. (Tell me if you try it and the code looks worse)

Reply:
@ashb I do agree that the code would look a lot cleaner without the mock. I'll remove it.